scuttlebutt-rpc/src/main/java/net/consensys/cava/scuttlebutt/rpc/mux/RPCHandler.java

```diff
@@ -53,11 +48,17 @@
   private final Runnable connectionCloser;
   private final ObjectMapper objectMapper;
 
-  private Map<Integer, CompletableAsyncResult<RPCMessage>> awaitingAsyncResponse = new HashMap<>();
+  private Map<Integer, CompletableAsyncResult<RPCResponse>> awaitingAsyncResponse = new HashMap<>();
   private Map<Integer, ScuttlebuttStreamHandler> streams = new HashMap<>();
 
   private boolean closed;
 
+  /**
+   * We run each each update on this executor to update the request state synchronously, and to handle the underlying
+   * connection closing by failing the in progress requests and not accepting future requests
+   */
+  private final ExecutorService executor = Executors.newSingleThreadExecutor();
```

Comment:
Is this executor closed? If vert.x is in use, you may be able to leverage it for this purpose.